crates/wit-parser/src/ast/resolve.rs

```diff
@@ -292,6 +302,30 @@ impl<'a> Resolver<'a> {
                     Ok(())
                 })
                 .unwrap();
+            decl_list
+                .for_each_nest(|nest: &Nest| {
+                    let deps = foreign_deps
+                        .entry(nest.id.package_name())
+                        .or_insert_with(|| {
+                            self.foreign_dep_spans.push(nest.id.span);
+                            IndexMap::new()
+                        });
+                    let id = *deps.entry(nest.name.name).or_insert_with(|| {
+                        log::trace!(
+                            "creating an interface for foreign dep: {}/{}",
+                            nest.id.package_name(),
+                            nest.name.name
+                        );
+
+                        AstItem::Interface(self.alloc_interface(nest.name.span))
+                    });
+                    match id {
+                        AstItem::Interface(id) => foreign_interfaces.insert(id),
+                        AstItem::World(_) => unreachable!(),
+                    };
+                    Ok(())
+                })
+                .unwrap();
```

Comment:
I think ideally if nest used a `UsePath` then thiscould go away, along with the `for_each_nest` method?